### ISSUE TITLE
[BUG] Application template generation is not functional

### ISSUE THREAD

**Issue by dizzzz** [eXist-db/eXide#251]
**Describe the bug**
Generating a new application does not work. Running the app results in

![image](https://user-images.githubusercontent.com/1700062/79685411-fe08f200-8238-11ea-8370-a08c62607bec.png)

Observation: The modules and resources directory are empty 

**Expected behavior**
Execute Application - New Application should yield in a working app.

**To Reproduce**
Run in eXide , do Application - New Application, follow wizard, open app

**Unit Test**
N/A

**Screenshots**

**Context (please always complete the following information):**
 - OS: [e.g. macOS 10.14.6] macOS 10.15.4
 - eXist-db Version: [e.g. 5.1.1] 5.3/snapshoy
 - Java Version: [e.g. Java8u121] Java11
-  App Version: [e.g. 1.0.0]  2.4.9

**Additional context**
N/A

**Comment by lguariento** [comment]
This _seems_ not to be related to https://github.com/eXist-db/shared-resources/pull/48. Reverting the `Jquery-utils.js` to how it was before the pull request yields the same issue.

**Comment by duncdrum** [comment]
we could use cypress to test exides app generator.

**Comment by lguariento** [comment]
<i>Opens https://www.cypress.io/ on the browser and browses around to see how it works</i>

**Comment by joewiz** [comment]
@dizzzz What values did you supply in the app wizard?

**Comment by dizzzz** [comment]
@joewiz nothing special after retrying in iterations multiple options. Just fill in 'aaa' and 'http://aaa" where applicable. Does it work for you then?

**Comment by joewiz** [comment]
@dizzzz Any particular template, or did you try all 3?

No, I haven’t tried yet - I’m just trying to tease out a consistent set of steps we can follow in common to test the feature.

**Comment by lguariento** [comment]
@joewiz I tried all templates and quite a few different names/titles, the generator consistently generates a new application with `modules` and `resources` empty on 5.3, with or without the new JS file.

**Comment by dizzzz** [comment]
I tried only the eXist-db template,,,,

**Comment by joewiz** [comment]
Ok, here are the exact steps I just followed for using the New application wizard in eXide. These steps successfully produced an app with the modules and resources collections populated, when using eXist 5.2.0 with shared-resources 0.8.5 and eXide 2.4.9 (but I'm able to reproduce your results in eXist 5.3.0-DEVELOP).

1. Select eXide's "Application" menu > "New application"
2. Complete screen 1:
    > ![Screen Shot 2020-04-20 at 6 16 13 PM](https://user-images.githubusercontent.com/59118/79805171-8d92cb80-8333-11ea-8a08-e6289e194828.png)
3. Select the "Next" button.
4. Complete screen 2:
    > ![Screen Shot 2020-04-20 at 6 16 22 PM](https://user-images.githubusercontent.com/59118/79805248-b61ac580-8333-11ea-9701-bd337ee60a48.png)
5. Select the "Next" button.
6. Complete screen 3:
    > ![Screen Shot 2020-04-20 at 6 16 31 PM](https://user-images.githubusercontent.com/59118/79805274-c29f1e00-8333-11ea-9df1-4186145638d7.png)
7. Select the "Done" button. This triggers a POST request to eXide's modules/deployment.xql, with the following parameters:
    ```
    collection: /db/apps//
    template: existdb
    type: application
    target: my-app
    name: http://joewiz.org/ns/my-app
    abbrev: my-app
    title: My App
    version: 0.1
    status: alpha
    prepare: pre-install.xql
    finish: 
    author: 
    description: 
    website: 
    owner: 
    password: 
    group: 
    mode: rw-rw-r--
    ```
8. Dismiss the Open Document pane.
9. Browse the DB sidebar, and note that the modules and resources folders are full:
    > ![Screen Shot 2020-04-20 at 6 17 53 PM](https://user-images.githubusercontent.com/59118/79805755-d8611300-8334-11ea-8fba-0cfdc7a6c4c8.png)
10. Navigate to `http://localhost:8080/exist/apps/my-app/index.html`, and the app opens:
    > ![Screen Shot 2020-04-20 at 6 25 16 PM](https://user-images.githubusercontent.com/59118/79805539-5ec92500-8334-11ea-9554-44bffc90b5ea.png)

So I'm unable to reproduce the bug in 5.2.0. But in eXist 5.3.0-DEVELOP my experience matches yours.

I just wrote this to @lguariento in Slack: 

> When you run the eXide new application wizard, the form fields are sent in a POST request to `modules/deployment.xql` . The last change to that file was included in eXide 2.4.8, from last April: https://github.com/eXist-db/eXide/commit/e39560ad6caa88e6c96e7ef7ed7038b18a5a058d. The current release is eXide 2.4.9, from last August, but I’m not able to reproduce the problem you all reported when running eXist 5.2.0. This makes me think that the problem lies outside of eXide, and is a problem with some other aspect of eXist 5.3.0-develop or the other apps included in builds thereof.

**Comment by lguariento** [comment]
Sorry if I wasn't clear - I actually experience this issue only with eXist 5.3.0-DEVELOP. Up to 5.2 it all works fine, as confirmed by the fact that I was testing my https://github.com/eXist-db/shared-resources/pull/48 with 5.2 indeed.

**Comment by adamretter** [comment]
@joewiz @lguariento if you think this is a regression in 5.3.0-SNAPSHOT, is there anything relevant in `exist.log`?

**Comment by lguariento** [comment]
Unfortunately I can't see anything interesting in the logs. Also, I notice that my JS (https://github.com/eXist-db/shared-resources/pull/48) has not yet included in the 5.3.0-DEVELOP, so that's definitely not an issue.

**Comment by joewiz** [comment]
@lguariento What version of shared resources are you running? My local build of eXist picks up the current release, v0.8.5, which contains your fix: https://github.com/eXist-db/shared-resources/releases/tag/v0.8.5

**Comment by lguariento** [comment]
@joewiz Good point. My The 5.3 on Docker (MacOS), pulling existdb/existdb:latest, picks up release 0.8.2.

**Comment by lguariento** [comment]
@joewiz No, sorry, 0.8.5. But my edited JS is not there (indeed there's no check if you put an empty title/name/target).

**Comment by joewiz** [comment]
@lguariento That's strange. I wonder if the Docker build is stuck somehow. v0.8.5 is available in the [public-repo](http://exist-db.org/exist/apps/public-repo/packages/shared.html?eXist-db-min-version=3.0.4), so I'd expect all builds to pick it up. 

By the way, you were definitely clear! I started by testing in 5.2.0, since I couldn't imagine anything in 5.3.0 causing a regression. But then I noticed that you and Dannes said that you were using 5.3.0, so I tested with that, and then realized I could see the problem with 5.3.0.

**Comment by lguariento** [comment]
@joewiz sorry, I cleared all cache LocalStorage and all, and I'm now loading the correct JS, it was a cache issue, all fine (in terms of updated shared resources.)

SO, I now get a new interesting error then: when clicking 'done', after filling in the fields in the new-application-generator, I get this empty message:
![Screenshot_2020-04-21 eXide](https://user-images.githubusercontent.com/10143842/79871550-680dcc80-83dc-11ea-8ef0-2f3106d38896.png)

```
Request URL:http://localhost:8080/exist/apps/eXide/modules/deployment.xql
Request method:POST
Remote address:[::1]:8080
Status code:
500
Version:HTTP/1.1
Referrer Policy:no-referrer-when-downgrade
```

The form data contains this:

```
collection | "/db/apps//"
```

This is a problem of course, because it seems like eXist-db doesn't pick up the name of the application.

**Edit**

The error 500 was due to my not being logged in. After logging in the behaviour is again what experienced earlier. I noticed, though, that the FormData sent always contains `collection	"/db/apps//"`, even in 5.2 (where everything works as expected).
